### PR TITLE
[SYCL] test for https://github.com/intel/llvm/pull/4405

### PR DESCRIPTION
**Pull request by smaslov-intel**
Verify balanced event retains/releases under special circumstances.
Signed-off-by: Sergey V Maslov <sergey.v.maslov@intel.com>

### CODE REVIEW

**Comment by romanovvlad** [comment]
Could you please add more information about the bug this test should trigger?

**Comment by smaslov-intel** [comment]
> Could you please add more information about the bug this test should trigger?

It was crashing due to imbalance of event retain/releases (from within Level-Zero plugin) with this special test. I've updated PR description too.

**Comment by asudarsa** [comment]
It might be helpful to add the description of this test as a comment inside the test.

**Comment by romanovvlad** [comment]
> 
> 
> > Could you please add more information about the bug this test should trigger?
> 
> It was crashing due to imbalance of event retain/releases (from within Level-Zero plugin) with this special test. I've updated PR description too.

Thanks. Can we have a test which makes that event retain/release is balanced now in a more reliable way than just crash/not crash? I believe there were are checks in the Level Zero plugin that track reference counters.

**Comment by smaslov-intel** [comment]
The crash is actually a _reliable_ "die" in L0 plugin, when release is called on an object that has 0 ref-count. I wouldn't want to spend time on crafting a more general test here for the reported regression.

**Comment by romanovvlad** [comment]
> 
> 
> The crash is actually a _reliable_ "die" in L0 plugin, when release is called on an object that has 0 ref-count. 

Will the test crash if the number of "retains" is bigger than the number of "retains"(imbalance in the opposite direction)?

> I wouldn't want to spend time on crafting a more general test here for the reported regression.

Ok.

File: SYCL/Regression/host_unified_memory.cpp [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+int main(int argc, char *argv[]) {
+  queue Q;
+  auto BE =
+      (bool)(Q.get_device()
+                 .template get_info<sycl::info::device::opencl_c_version>()
+                 .empty())
+          ? "L0"
+          : "OpenCL";
+  device dev = Q.get_device();
+  size_t max_compute_units = dev.get_info<info::device::max_compute_units>();
```
Comment (romanovvlad) [inline SYCL/Regression/host_unified_memory.cpp:23]: Minor.
Information about compute units seems to be irrelevant to the test case.

**Comment by smaslov-intel** [comment]
> Will the test crash if the number of "retains" is bigger than the number of "retains"(imbalance in the opposite direction)?

Not this one, but we run entire LIT testing with ZE_DEBUG=4, mode in which L0 resources leaks are detected, we also run tests with valgrind. This test is specifically for the regression observed (and not caught by any other existing testing).

**Review by romanovvlad — APPROVED** [review]
Approving. Please, note there was a minor comment from me.
